test/lib/array.jl

```diff
@@ -6,3 +6,7 @@ using Zygote: ZygoteRuleConfig
 
 test_rrule(ZygoteRuleConfig(), x->sum(sin, Diagonal(x)), ones(2); rrule_f=rrule_via_ad, check_inferred=false)
 test_rrule(ZygoteRuleConfig(), x->sum(sin, Diagonal(x)), rand(3); rrule_f=rrule_via_ad, check_inferred=false)
+
+# issue 44179
+
+@test Zygote._restore([nothing, 1, nothing], Val(3)) == (nothing, 1, nothing)
```

Comment:
The change looks safe to me. 

But can you replace this test with a `gradient` example (or a few) which failed before? I think we should mostly test the end result, not internal functions. If someone re-writes the rule for `map`, they are free to delete `_restore`, but shouldn't break actual uses.